server/remote/msgraph/get_me.go

```diff
@@ -3,13 +3,31 @@
 
 package msgraph
 
-import "github.com/mattermost/mattermost-plugin-mscalendar/server/remote"
+import (
+	"errors"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/remote"
+)
 
 func (c *client) GetMe() (*remote.User, error) {
 	graphUser, err := c.rbuilder.Me().Request().Get(c.ctx)
 	if err != nil {
 		return nil, err
 	}
+
+	if graphUser.ID == nil {
+		return nil, errors.New("User has no ID")
+	}
+	if graphUser.DisplayName == nil {
+		return nil, errors.New("User has no Display Name")
+	}
+	if graphUser.UserPrincipalName == nil {
+		return nil, errors.New("User has no Principal Name")
+	}
+	if graphUser.Mail == nil {
+		return nil, errors.New("User has no mail")
```

Comment:
On second thought, I think we should say something like this:
```suggestion
		return nil, errors.New("User has no email address. Please connect the account to Outlook.")
```

Also, does this error show to the user after returning?

Reply:
GetMe, at the moment, is only used on the `CompleteOAuth2` function. This function returns an error that eventually will be processed by `WriteJSONError`, which will write the error on the web response, showing it to the user.
That being said, maybe the message should be more clear, since the may already think they are "connecting the account to Outlook". This error, if I remember correctly, is because the microsoft account is not connected to an outlook account, right?
What do you think about "User has no email address. Make sure the Microsoft account is associated to an Outlook product."

Reply:
> "User has no email address. Make sure the Microsoft account is associated to an Outlook product."

Sounds good to me!